app/helpers/search_helper.rb

```diff
@@ -170,4 +204,44 @@ def abbreviated_text(words)
 
     abbreviated_words
   end
+
+  # Similar to `abbreviated_text`, but considers HTML tags and keeps them intact
+  def abbreviated_html(html, max_length: 1200)
+    doc = Nokogiri::HTML::DocumentFragment.parse(html)
+    text_length = 0
+
+    doc.traverse do |node|
+      next unless node.text?
+
+      original_text = node.content
+      abbreviated_text = process_text_node(original_text, text_length, max_length)
+      abbreviated_text = preserve_spaces(original_text, abbreviated_text) || ""
+
+      text_length += abbreviated_text.length
+      node.replace(Nokogiri::XML::Text.new(abbreviated_text, doc))
+
+      break if text_length >= max_length
+    end
+
+    doc.to_html
+  end
+
+  def process_text_node(content, current_length, max_length)
+    return content if current_length >= max_length
+
+    truncated_text = truncate_formatted_text(content, length: 100)
+
+    if current_length + truncated_text.length > max_length
+      t = truncated_text[0..(max_length - current_length - 1)]
+      truncated_text = "#{t}..."
```

Comment:
I believe we could use the [truncate](https://apidock.com/rails/v7.1.3.4/String/truncate) rails string extension:

```suggestion
      truncated_text.truncate(max_length - current_length - 1)
```
I am also wondering if the `truncate_formatted_text` is required, since we are processing the text nodes here.

Reply:
Thank you, I have added `truncate` 👏🏻 

The `truncate_formatted_text` call is required to shorten additional portions of the search results, so I left it in for now.